pkg/kubectl/cmd/run.go

```diff
@@ -213,7 +214,9 @@ func Run(f cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cobr
 		}
 		switch restartPolicy {
 		case api.RestartPolicyAlways:
-			if contains(resourcesList, v1beta1.SchemeGroupVersion.WithResource("deployments")) {
+			if contains(resourcesList, appsv1beta1.SchemeGroupVersion.WithResource("deployments")) {
+				generatorName = "deployment/apps.v1beta1"
+			} else if contains(resourcesList, extensionsv1beta1.SchemeGroupVersion.WithResource("deployments")) {
 				generatorName = "deployment/v1beta1"
```

Comment:
Well, changing default generator is a breaking change, so I can agree with that one. But applying new defaults if new generator was specified against old server seems wrong and will unnecessarily complicate the code in those places (I'm talking both about run and create generators here). I'll leave default generators as is, we'll change that when we deprecate extensions/v1beta1.Deployments (maybe in 1.7) and the old generator.